src/pretix/presale/views/event.py

```diff
@@ -104,6 +108,29 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class EventIcalDownload(EventViewMixin, View):
+    def get(self, request, *args, **kwargs):
+        if not self.request.event:
+            raise Http404(_('Unknown event code or not authorized to access this event.'))
+
+        cal = Calendar()
+        cal.add('version', '2.0')
+        cal.add('prodid', '-//{}//{}//'.format(self.request.event.organizer.name, self.request.event.name))
+
+        event = Event()
+        event.add('summary', self.request.event.name)
+        event.add('dtstart', self.request.event.date_from)
+        event.add('dtend', self.request.event.date_to)
+        event.add('dtstamp', datetime.now(pytz.utc))
+        event.add('location', self.request.event.location)
```

Comment:
@rixx thanks for your explanation. I think I should also warp `event.name`(summary) too coz it is also `I18nCharField`.